dataflow/extensible-templates/dataflow_udf_transform.js

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START dataflow_extensible_template_udf]
+/**
+ * User-defined function (UDF) to transform events
+ * as part of a Dataflow template job.
+ *
+ * @param {string} inJson input Pub/Sub JSON message (stringified)
+ */
+ function process(inJson) {
+    var obj = JSON.parse(inJson),
```

Comment:
It looks like [Nashorn (partially) supports ECMA6](https://openjdk.org/jeps/292), so let's see if we can modernize this Javascript a little bit.

Does it work if we change `var` to `const`? This forces the "variable" to be immutable, and makes it respect naming scopes.

Reply:
No, it doesn't unfortunately. Tests failed after I made the switch. Keeping `var` but we should file a bug.

Reply:
Yeah, let's keep `var` for now. Although if Nashorn is getting deprecated in the next Java version, maybe we can let it be. It looks like Java 11 will end of life in around 1 year and alongside with that [Nashorn will be deprecated](https://www.graalvm.org/22.2/reference-manual/js/NashornMigrationGuide/), the templates and other UDFs will most likely have to adapt GraalVM or something similar since Nashorn won't be around.